CI is failing with LinearStretch.__init__() got an unexpected keyword argument 'out'

I restarted a previously successful build on `main` and now it fails with the following:

```
        # Copied from the __call__ internals of glue/viewers/image/composite_array.py
        data = interval(x)
        data = contrast_bias(data, out=data)
>       data = stretch(data, out=data)
E       TypeError: LinearStretch.__init__() got an unexpected keyword argument 'out'

jdaviz\configs\default\plugins\plot_options\plot_options.py:862: TypeError
```

Example log: https://github.com/spacetelescope/jdaviz/actions/runs/7088856608/job/19350503369

Main suspect:

* https://github.com/glue-viz/glue/pull/2453 (released in https://github.com/glue-viz/glue/releases/tag/v1.17.0)

[🐱](https://jira.stsci.edu/browse/JDAT-4035)

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -65,6 +65,8 @@ Other Changes and Additions
 Bug Fixes
 ---------
 
+- Compatibility with glue-core 1.17. [#2591]
+
 Cubeviz
 ^^^^^^^
 

--- a/jdaviz/configs/default/plugins/plot_options/plot_options.py
+++ b/jdaviz/configs/default/plugins/plot_options/plot_options.py
@@ -4,6 +4,7 @@ import os
 import matplotlib
 import numpy as np
 
+from astropy.utils import minversion
 from astropy.visualization import (
     ManualInterval, ContrastBiasStretch, PercentileInterval
 )
@@ -31,6 +32,8 @@ from jdaviz.core.custom_traitlets import IntHandleEmpty
 from scipy.interpolate import make_interp_spline
 
 __all__ = ['PlotOptions']
+
+GLUE_LT_1_17 = not minversion("glue", "1.17")
 
 
 class SplineStretch:
@@ -89,7 +92,10 @@ class SplineStretch:
 
 # Add the spline stretch to the glue stretch registry if not registered
 if "spline" not in stretches:
-    stretches.add("spline", SplineStretch(), display="Spline")
+    if GLUE_LT_1_17:
+        stretches.add("spline", SplineStretch(), display="Spline")
+    else:
+        stretches.add("spline", SplineStretch, display="Spline")
 
 
 @tray_registry('g-plot-options', label="Plot Options")
@@ -846,7 +852,10 @@ class PlotOptions(PluginTemplateMixin):
         # procedure in glue's CompositeArray:
         interval = ManualInterval(self.stretch_vmin_value, self.stretch_vmax_value)
         contrast_bias = ContrastBiasStretch(self.image_contrast_value, self.image_bias_value)
-        stretch = stretches.members[self.stretch_function_value]
+        if GLUE_LT_1_17:
+            stretch = stretches.members[self.stretch_function_value]
+        else:
+            stretch = stretches.members[self.stretch_function_value]()
         layer_cmap = layer.state.cmap
 
         # show the colorbar

